### PR TITLE
Add safe creation functions

### PR DESCRIPTION
**Pull request by fedgiac**
Adds two function to create the calldata necessary to create new Gnosis Safes. One creates deterministic addresses, the other creates standard addresses.

They will be used in the follow-up PR that will create the data necessary for the proposal to be executed in the Gnosis DAO transaction. Most notably, these functions will be used to create the Cow DAO.

### Test Plan

New unit tests.

I also manually verified that the bytecode is the same as the bytecode used by the Gnosis Safe frontend when creating a new safe.

### CODE REVIEW

File: test/safe.test.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  expect(proxies).to.have.length(1);
+  return proxies[0];
+}
+
+type DeployFunction = (
+  owners: Wallet[],
+  threshold: number,
+) => Promise<[TransactionResponse, GnosisSafeManager]>;
+
+function testSafe(deploy: DeployFunction) {
```
Comment (josojo) [inline test/safe.test.ts:35]: nit: Maybe testSafeSetup() would be a better name.

But I really like how you were writing this test!

File: src/ts/lib/safe.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+): Promise<string[]> {
+  const receipt = await response.wait();
+  const creationEvents = receipt.logs
+    .filter(({ address }) => address === proxyFactoryAddress)
+    .map((log) => proxyFactoryIface.parseLog(log))
+    .filter(({ name }) => name === "ProxyCreation");
+  return creationEvents.map(({ args }) => args.proxy);
+}
+
+export async function getFallbackHandler(
```
Comment (josojo) [inline src/ts/lib/safe.ts:111]: This is hard to review. Did you copy the code from somewhere? maybe you can provide a source

File: src/tasks/ts/safe.ts [hunk 3]
```diff
@@ -6,22 +6,28 @@ import {
   MetaTransaction,
 } from "@gnosis.pm/safe-contracts";
 import GnosisSafe from "@gnosis.pm/safe-contracts/build/artifacts/contracts/GnosisSafe.sol/GnosisSafe.json";
-import GnosisSafeProxyFactory from "@gnosis.pm/safe-contracts/build/artifacts/contracts/proxies/GnosisSafeProxyFactory.sol/GnosisSafeProxyFactory.json";
+import CompatibilityFallbackHandlerDeployment from "@gnosis.pm/safe-deployments/src/assets/v1.3.0/compatibility_fallback_handler.json";
```
Comment (josojo) [inline src/tasks/ts/safe.ts:9]: Was there any official requirement for the fallback handler? I can not remember any discussion.

**Review by josojo — APPROVED** [review]
Very nice

File: src/ts/lib/safe.ts (hunk 2 above)
Comment (fedgiac) [reply src/ts/lib/safe.ts:112]: I added a comment with some references and made the byte decoding more readable. Do you think it makes sense now?

File: src/tasks/ts/safe.ts (hunk 3 above)
Comment (fedgiac) [reply src/tasks/ts/safe.ts:9]: There was no official requirement. I was discussing with Auryn and he mentioned that this is needed for receiving ERC1155 tokens, which I thought was good to have. Will start a discussion in #dex-token.

File: src/ts/lib/safe.ts (hunk 2 above)
Comment (josojo) [reply src/ts/lib/safe.ts:112]: Yes, much much better! Now it's easy to understand. Looks good

It's a small pity, that `0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5` was not exported somewhere in their project.

**Pull request merged**